### ISSUE TITLE
Rules used from first dependency container

### ISSUE THREAD

**Issue by yehudahkay** [epartment/nova-dependency-container#81]
I am using the following code to allow me to display different institutions based on the user type selected. However the second field uses the first fields rules which means that the rule fails

```
$fields[] = NovaDependencyContainer::make([
                Select::make('Institution', 'institution_id')
                    ->options($trusts)
                    ->rules([function ($attribute, $value, $fail) use ($trusts) {
                        if (!array_key_exists($value, $trusts)) {
                            $fail($attribute . ' is invalid.');
                        }
                    }])
            ])->dependsOn('privileges', '{"role":"indexer_triage"}')->dependsOn('privileges', '{"role":"referral_management"}');
            $fields[] = NovaDependencyContainer::make([
                Select::make('Institution', 'institution_id')
                    ->options($departments)
                    ->rules([function ($attribute, $value, $fail) use ($departments) {
                        if (!array_key_exists($value, $departments)) {
                            $fail($attribute . ' is invalid.');
                        }
                    }])
            ])->dependsOn('privileges', '{"role":"consultant"}')->dependsOn('privileges', '{"role":"departmental_triage"}');
```

**Comment by wize-wiz** [comment]
There is a PR for probably the same issue [#60](https://github.com/epartment/nova-dependency-container/pull/60). This is probably the only PR I'm going to merge the coming days, maybe I'll have some time tonight to push this one in a new release.

**Comment by wize-wiz** [comment]
@yehudahkay Should be fixed in [1.2.4](https://github.com/epartment/nova-dependency-container/releases/tag/1.2.4), could you please verify and if so close this issue?

**Comment by yehudahkay** [comment]
Yes thank you!
